api/rooms.js

```diff
@@ -0,0 +1,48 @@
+const express = require("express");
+
+module.exports = function rooms(db) {
+  const router = express.Router();
+
+  const usersCollection = db.collection("users");
+  const roomsCollection = db.collection("rooms");
+
+  // GET
+  router.get("/", function (req, res) {
+    roomsCollection.find().toArray((err, userDocs) => {
+        res.send({ data: userDocs.map(encodeRoom) });
+      });
+  });
+
+  // POST
+  router.post("/", function (req, res, next) {
+    const userIds = req.body.participants;
```

Comment:
You need to declare this with `let` because you reassign this below - you can't reassign to a `const` variable.

When you try to reassign this variable in the `try` below, an exception is thrown.  Since we are catching and ignoring errors, execution continues, and `userIds` is still an array of strings